src/connection/providers/browser/ethereum-provider/eip-6963.ts

```diff
@@ -0,0 +1,218 @@
+import { EIP1193Provider } from './eip-1193.js';
+
+/**
+ * Get a list of all detected EIP-6963 providers (browser wallets).
+ */
+export async function getEthereumProviders (): Promise<EIP6963ProviderDetail[]> {
+
+    const providers = await detectProviders();
+
+    return providers;
+}
+
+/**
+ * Get the EIP-6963 provider (browser wallet) that matches the given identifier.
+ *
+ * @remarks
+ * On page reload, the `uuid`s of the EIP-6963 providers change, meaning we cannot use them to identify wallets
+ * over multiple sessions. To alleviate this, we use the `name` and `rdns` properties to provide a 'fuzzy' match
```

Comment:
This is interesting that the uuid's change? Is there a rational youve understood for why? Wouldnt a wallet just have a consistent universal id?

Reply:
Also 👍 for the fuzzy workaround.

I think one thing to keep in mind is that really if someone installs a malicious extension theyre likely just screwed in so many ways already as well

Reply:
> This is interesting that the uuid's change? Is there a rational youve understood for why? Wouldnt a wallet just have a consistent universal id?

I haven't found a specific rationale for this, but basically the EIP-6963 spec states, that provider uuid's actually identify "provider sessions [...] during the lifetime of the page.". So my best guess is, that there's no mechanism/service for browser wallet extensions to register for a globally unique id, so they scoped this down to a session id that's generated on the fly. Not _super_ useful really...